### PR TITLE
Friends App  by Roophee

### PR DESCRIPTION
**Pull request by Roophee**
### Friends App

[Demo](https://roophee.github.io/friends-app/)
[Code base](https://github.com/Roophee/frontend-2021-homeworks/tree/friends-app-old-school/submissions/Roophee/friends-app)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/Roophee/friends-app/index.js [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+        return filteredUsersData;
+    }else if(filterSettings.filteredData.length >= 1 && filterSettings.secondFilteredData.length < 1 && filterSettings.filterOne === filter.name){
+        filteredUsersData = usersData.filter((user) => filter(user));
+        filterSettings.filteredData = filteredUsersData;
+        filterSettings.filterOne = filter.name;
+        return filteredUsersData;
+    }else if(filterSettings.filteredData.length >= 1 && filterSettings.secondFilteredData.length < 1 && filterSettings.filterOne !== filter.name){
+        filteredUsersData = filterSettings.filteredData.filter((user) => filter(user));
+        filterSettings.secondFilteredData = filteredUsersData;
+        filterSettings.filterTwo = filter.name;
```
Comment (babayK0) [inline submissions/Roophee/friends-app/index.js:112]: i don't see this props in initial `filterSettings`

`filterOne`
`filterTwo`

and what is the difference between these ones?

**Review by babayK0 — CHANGES_REQUESTED** [review]
Cool design ! I like it   :sunglasses:

First of all i want ask you why all filters resets when i start type in search input ? 
Also watch my comments below

File: submissions/Roophee/friends-app/index.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+const DATA_URL = 'https://randomuser.me/api/?results=100';
+const RAW_DATA = [];
+const USER_STOR = new Map();
+const mainSection = document.querySelector('.main__section');
+const filterSettings = {searchParam: '',
+                        genderParam: '',
+                        ageParam: '',
+                        nameParam: '',
+                        filteredData: [],
+                        secondFilteredData: []};
```
[the remark this answers is not in the document]
Comment (babayK0) [reply submissions/Roophee/friends-app/index.js]: @Roophee can you answer please ?
why these variables called like this 
not obvious

File: submissions/Roophee/friends-app/index.js [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
+genderParam: '',
+ageParam: '',
+nameParam: '',
+filteredData: [],
+secondFilteredData: [],
+filterOne: '',
+filterTwo: '',
+};
+const wasFirstFiltration = () => filterSettings.filteredData.length >= 1;
+const wasSecondFiltration = () => filterSettings.secondFilteredData.length >= 1;
```
Comment (babayK0) [inline submissions/Roophee/friends-app/index.js:18]: 1. verb
1. `wasFiltration` ? maybe `getIsFiltered` will be better name for this one
1. if you use functions you can do things like this
```suggestion
const getIsFiltered = (data) => data.length >= 1;
```
or 
```suggestion
const getIsFiltered = (settingDataKey) => filterSettings[settingDataKey].length >= 1;
```

File: submissions/Roophee/friends-app/index.js [hunk 4]
```diff
@@ -1,15 +1,23 @@
...
+const thisFilterMadeFirstFiltration = (filter) => filterSettings.filterOne === filter.name;
+const thisFilterMadeSecondFiltration = (filter) => filterSettings.filterTwo === filter.name;
```
Comment (babayK0) [inline submissions/Roophee/friends-app/index.js:20]: 1. verb
2. 
```suggestion
const checkWichFiltrationWasMade = (filter, comparativeFilterName) => comparativeFilterName === filter.name;

```
or
```suggestion
const checkWichFiltrationWasMade = (filter, filterKey) => filterSettings[filterKey] === filter.name;

```

File: submissions/Roophee/friends-app/index.js [hunk 5]
```diff
@@ -94,57 +102,85 @@ const nameSort = (prev, next) => {
...
     let sortedUsersData = [];
-    if (filterSettings.filteredData.length >= 1 && filterSettings.secondFilteredData.length >= 1) {
+
+    const thisSortFunctionByAge = (sortFunction) => {
+        return sortFunction.name === 'ageSort'
+    }
+
+    const thisSortFunctionByName = (sortFunction) => {
+        return sortFunction.name === 'nameSort'
+    }
```
Comment (babayK0) [inline submissions/Roophee/friends-app/index.js:156]: 1. verb
2. make more dry please) `const foo = (a , b) => a === b`

File: submissions/Roophee/friends-app/index.js [hunk 6]
```diff
@@ -94,57 +102,85 @@ const nameSort = (prev, next) => {
...
+
+    const sortByAgeAscending = () => {
+        return filterSettings.ageParam === 'ascending';
+    }
+
+    const sortByNameAlphabetically = () => {
+        return filterSettings.nameParam === 'az';
+    }
```
Comment (babayK0) [inline submissions/Roophee/friends-app/index.js:164]: 1. verbs
2. dry

File: submissions/Roophee/friends-app/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Roophee) [reply submissions/Roophee/friends-app/index.js]: @babayK0 I was trying to implement filtering an array with two filters. The first filter stores the filtered result in a `filteredData: []` array. The second filter make filtration with the data after the first filter and saves it to the second array `secondFilteredData: []`  to display it on the screen, and then the method is applied to get Nodes from NODE_STORAGE.

File: submissions/Roophee/friends-app/index.js [hunk 7]
```diff
@@ -163,13 +199,21 @@ const uncheckAllRadio = () => {
...
     document.querySelectorAll('input[name="name"]').forEach((item) => {
         item.checked = false;
     });
+}
+
+const uncheckSortByAgeRadio = () => {
     document.querySelectorAll('input[name="age"]').forEach((item) => {
         item.checked = false;
     })
+}
```
Comment (babayK0) [inline submissions/Roophee/friends-app/index.js]:
```suggestion
const uncheckSortByRadio = (radioName) => {
    document.querySelectorAll('input[name=radioName]').forEach((item) => {
        item.checked = false;
    });
}

**Review by babayK0 — CHANGES_REQUESTED** [review]
Need more DRY :rofl:

File: submissions/Roophee/friends-app/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Roophee) [reply submissions/Roophee/friends-app/index.js]: @babayK0  Sorting methods are applied depending on whether the first filtration and / or the second filtration was performed. That is, the methods are applied to arrays where filtering results are stored.

[the remark this answers is not in the document]
Comment (babayK0) [reply submissions/Roophee/friends-app/index.js]: change names of this vars to more understandable

**Review by babayK0 — APPROVED** [review]
:+1:

**Pull request merged**